# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/PQuota.ipdl
# Commit: 396d9243d417
# Full Hash: 396d9243d4172aef258b24a09cf2a2574d86f87a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 09:46:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add a dedicated method for clearing storages of a specific client; r=dom-storage-reviewers,jari
#   
#   Client storages can be currently cleared by using the optional argument of
#   nsIQuotaManagerService::ClearStoragesForPrincipal. However, supporting clearing
#   storages for both a principal and a client is not ideal for several reasons.
# ==============================================================================

diff -r fe4a050ac4da -r 396d9243d417 dom/quota/PQuota.ipdl
--- a/dom/quota/PQuota.ipdl	Wed Oct 09 05:34:35 2024 +0000
+++ b/dom/quota/PQuota.ipdl	Wed Oct 09 05:34:36 2024 +0000
@@ -172,6 +172,11 @@
                                Type? clientType)
       returns(BoolResponse response);
 
+  async ClearStoragesForClient(PersistenceType? persistenceType,
+                               PrincipalInfo principalInfo,
+                               Type clientType)
+      returns(BoolResponse response);
+
   async ClearStoragesForOriginPrefix(PersistenceType? persistenceType,
                                      PrincipalInfo principalInfo)
       returns(BoolResponse response);